RHOAIENG-17008 - Additional MNMGPU feedback (#612)
opendatahub-io/opendatahub-documentation: modules/deploying-models-using-multiple-gpu-nodes.adoc

----
+

> . From the namespace where you would like to deploy the model, create a PVC for model storage. Provide the size of the model and the name of your storage class. The storage class must be file storage. For example:

Reviewer: I would change this:

> Provide the size of the model and the name of your storage class.

To something like this:

> Create a storage class using `Filesystem` `volumeMode` and the name of your storage class.  The storage size must be larger than the size of the model files on disk.

> . From the namespace where you would like to deploy the model, create a PVC for model storage. Create a storage class using `Filesystem volumeMode` and the name of your storage class. The storage size must be larger than the size of the model files on disk. For example:

Reviewer 2: ```suggestion
. From the namespace where you would like to deploy the model, create a PVC for model storage. Create a storage class using `Filesystem volumeMode`. Use this storage class for your PVC. The storage size must be larger than the size of the model files on disk. For example:
```

          value: <AWS endpoint> <7>
        - name: awsAnonymousCredential
          value: 'false'
        - name: AWS_DEFAULT_REGION
          value: <region> <8>
        - name: S3_VERIFY_SSL
          value: 'true' <9>
      volumeMounts:
        - mountPath: "/mnt/models/"
          name: model-volume
----
<1> The `chmod` operation is permitted only if your pod is running as root. Remove`chmod -R 777` from the arguments if you are not running the pod as root.
<2> Specify the path to the model.
ifndef::upstream[]
<3> The value for `containers.image`, located in your `InferenceService`. To access this value, run the following command: `oc get configmap inferenceservice-config -n redhat-ods-operator -oyaml | grep kserve-storage-initializer:`
endif::[]
ifdef::upstream[]
<3> The value for `containers.image`, located in your `donwload-model` container. To access this value, run the following command: `oc get configmap inferenceservice-config -n opendatahub -oyaml | grep kserve-storage-initializer:`
endif::[]
<4> The access key ID to your S3 bucket.
<5> The secret access key to your S3 bucket.
<6> The name of your S3 bucket.
<7> The endpoint to your S3 bucket.

> <8> The region for your S3 bucket.

Reviewer: The region option may confuse anyone that is not using an AWS S3 instance (which is most of our customers).

Maybe something like this:

> <8> The region for your S3 bucket if using an AWS S3 buckets.  If using other S3 compatible storage such as ODF or Minio, you can remove the AWS_DEFAULT_REGION environment variable.

<9> If you encounter SSL errors, change `S3_VERIFY_SSL` to `false`.
+

> . Create the `vllm-multinode-runtime` custom runtime:

Reviewer: I would avoid the term "custom runtime".  

Maybe something like this instead.

> . Create the `servingRuntime` with the name `vllm-multinode-runtime` :

Author: Thank you! I'm curious why we want to avoid the term "custom runtime"? I was under the impression that this procedure required adding a custom runtime. If that was the case, it seems helpful to mirror the wording in our documentation, where we document how to [add a custom runtime](https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2-latest/html-single/serving_models/index#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models). If this isn't the case, I can certainly rework, just want to make sure I understand it properly as well!

Reviewer: My avoidance of it is that "Custom runtime" doesn't really have a technical meaning.

We have "Supported Serving Runtime" templates that are shipped out of the box and an admin has the ability to setup their own Serving Runtime template.

Once a "custom" runtime template is created by an admin a user that is deploying a model server through the UI has no way to distinguish between the two in the UI.

In the case of this example, we are using an OOTB (Tech Preview) Serving Runtime Template to create an instance of the ServingRuntime for our model.  While we don't use the Dashboard to deploy it, we could do the exact same oc commands with a fully supported Serving Runtime Template to create a fully support vLLM Serving Runtime.  Is the multi-node Serving Runtime we create any more custom than the fully supported one?  

On a bit of an aside I have a bit of a vendetta against our usage of "Serving Runtime" in the product/docs (and some RFE's are in the works to hopefully to improve this).  We generally use the term "Serving Runtime" to mean both the templates that are available in the Administrator section which users see in the drop down menu when deploying with the UI and also the `ServingRuntime` object that is created from those templates when the user creates a model server.  IMO we should be referring to the `OpenShift Template` objects which we use to template a `ServingRuntime` as a "Serving Runtime Template" and the k8s objected created by those templates with `Kind: ServingRuntime` as "Serving Runtimes".

Back to the discussion of "Custom Runtimes" the link you provided I think it makes sense to use the term "Customer Runtime Template" in the context of adding a Runtime Template that is not available OOTB.  Once you deploy an instance of any template though, at what point does it become custom?

Author: Thanks for the explanation, @strangiato ! I think this discussion is beyond the scope of this doc but I will bring it up next time I speak with my PM. For now I will leave the wording as-is and keep you posted if I change anything. Thanks again!
